components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -140,6 +157,32 @@ open class GleanInternalAPI internal constructor () {
         return uploadEnabled
     }
 
+    /**
+     * Clear any pending metrics when telemetry is disabled.
+     */
+    fun clearMetrics() = Dispatchers.API.launch {
+        // There is only one metric that we want to survive after clearing all metrics:
+        // firstRunDate. Here, we store its value so we can restore it after clearing
+        // the metrics.
+        val firstRunDateMetric = GleanInternalMetrics.firstRunDate
+        val existingFirstRunDate = DatetimesStorageEngine.getSnapshot(
+            firstRunDateMetric.sendInPings[0], false)?.get(firstRunDateMetric.identifier)?.let {
+            parseISOTimeString(it)
+        }
+
+        pingStorageEngine.clearPendingPings()
+        storageEngineManager.clearAllStores()
+
+        // Store a "dummy" KNOWN_CLIENT_ID in clientId.  This will make it easier to detect pings
+        // that were sent unintentionally when uploading is disabled.
```

Comment:
nit: can we slightly rephrase this to "This will make it easier to detect if pings are unintentionally sent after upload is disabled"